airflow-core/src/airflow/ui/src/pages/Events/EventsFilters.tsx

```diff
@@ -0,0 +1,226 @@
+import { useCallback } from "react";
+import { useTranslation } from "react-i18next";
+import { useSearchParams } from "react-router-dom";
+
+import { useTableURLState } from "src/components/DataTable/useTableUrlState";
+import { SearchParamsKeys } from "src/constants/searchParams";
+
+import { ResetButton } from "./ResetButton";
+import { getFilterCount, formatDateTimeLocalValue } from "./filterUtils";
+
+const {
+  AFTER: AFTER_PARAM,
+  BEFORE: BEFORE_PARAM,
+  DAG_ID: DAG_ID_PARAM,
+  EVENT_TYPE: EVENT_TYPE_PARAM,
+  MAP_INDEX: MAP_INDEX_PARAM,
+  RUN_ID: RUN_ID_PARAM,
+  TASK_ID: TASK_ID_PARAM,
+  TRY_NUMBER: TRY_NUMBER_PARAM,
+  USER: USER_PARAM,
+} = SearchParamsKeys;
+
+export const EventsFilters = () => {
+  const { t: translate } = useTranslation(["browse", "common"]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { setTableURLState, tableURLState } = useTableURLState();
+
+  const { pagination, sorting } = tableURLState;
+
+  const resetPagination = useCallback(() => {
+    setTableURLState({
+      pagination: { ...pagination, pageIndex: 0 },
+      sorting,
+    });
+  }, [pagination, setTableURLState, sorting]);
+
+  // Extract filter values from URL
+  const afterFilter = searchParams.get(AFTER_PARAM) ?? "";
+  const beforeFilter = searchParams.get(BEFORE_PARAM) ?? "";
+  const dagIdFilter = searchParams.get(DAG_ID_PARAM) ?? "";
+  const eventTypeFilter = searchParams.get(EVENT_TYPE_PARAM) ?? "";
+  const mapIndexFilter = searchParams.get(MAP_INDEX_PARAM) ?? "";
+  const runIdFilter = searchParams.get(RUN_ID_PARAM) ?? "";
+  const taskIdFilter = searchParams.get(TASK_ID_PARAM) ?? "";
+  const tryNumberFilter = searchParams.get(TRY_NUMBER_PARAM) ?? "";
+  const userFilter = searchParams.get(USER_PARAM) ?? "";
+
+  const handleInputChange = useCallback(
+    (paramName: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+
+      if (value) {
+        searchParams.set(paramName, value);
+      } else {
+        searchParams.delete(paramName);
+      }
+      resetPagination();
+      setSearchParams(searchParams);
+    },
+    [resetPagination, searchParams, setSearchParams],
+  );
+
+  const handleDateTimeChange = useCallback(
+    (paramName: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+
+      if (value) {
+        // Convert datetime-local value to ISO string
+        try {
+          const date = new Date(value);
+
+          if (!isNaN(date.getTime())) {
+            searchParams.set(paramName, date.toISOString());
+          }
+        } catch {
+          // Invalid date, remove the parameter
+          searchParams.delete(paramName);
+        }
+      } else {
+        searchParams.delete(paramName);
+      }
+      resetPagination();
+      setSearchParams(searchParams);
+    },
+    [resetPagination, searchParams, setSearchParams],
+  );
+
+  const handleClearFilters = useCallback(() => {
```

Comment:
`onClearFIlter` does not cost much so bellow we get `onClearFilters={onClearFilters}` which is cleaner.